plugins/channel-safely.cpp

```diff
@@ -0,0 +1,480 @@
+        } else {
+            EM::registerListener(EventType::TICK, hoursHandler, plugin_self);
+            EM::registerListener(EventType::TICK, daysHandler, plugin_self);
+        }
+        EM::registerListener(EventType::JOB_INITIATED, jobStartHandler, plugin_self);
+        EM::registerListener(EventType::JOB_COMPLETED, jobCompletionHandler, plugin_self);
+        manager.manage_designations(out, true);
+        out.print("channel-safely enabled!\n");
+    } else if (!enable) {
+        EM::unregisterAll(plugin_self);
+        out.print("channel-safely disabled!\n");
+    }
+    enabled = enable;
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_onstatechange(color_ostream &out, state_change_event event) {
+    if(enabled && World::isFortressMode()) {
+        switch (event) {
+            case SC_UNKNOWN:
+                break;
+            case SC_WORLD_LOADED:
+                break;
+            case SC_WORLD_UNLOADED:
+                break;
+            case SC_MAP_LOADED:
+                manager.manage_designations(out, true);
+                break;
+            case SC_MAP_UNLOADED:
+                break;
+            case SC_VIEWSCREEN_CHANGED:
+                break;
+            case SC_CORE_INITIALIZED:
+                break;
+            case SC_BEGIN_UNLOAD:
+                break;
+            case SC_PAUSED:
+                break;
+            case SC_UNPAUSED:
+                break;
+        }
+    }
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_shutdown(color_ostream &out) {
+    namespace EM = EventManager;
+    EM::unregisterAll(plugin_self);
+    enabled = false;
+    return CR_OK;
+}
+
+command_result manage_channel_designations(color_ostream &out, std::vector<std::string> &parameters){
+    if(parameters.empty()){
+        manager.manage_designations(out, true);
+    } else if (parameters.size() == 1 && parameters[0] == "enable") {
+        return plugin_enable(out, true);
+    } else if (parameters.size() == 1 && parameters[0] == "disable") {
+        return plugin_enable(out, false);
+    } else if (parameters.size() == 1 && parameters[0] == "debug") {
+        debug_out = &out;
+        manager.debug();
+        debug_out = nullptr;
+    } else {
+        return CR_FAILURE;
+    }
+    return CR_OK;
+}
+
+void onNewHour(color_ostream &out, void* tick_ptr){
+    if(enabled && World::isFortressMode()) {
+        static int32_t last_tick_counter = 0;
+        int32_t tick_counter = (int32_t) ((intptr_t) tick_ptr);
+        if ((tick_counter - last_tick_counter) >= hourTicks) {
+            last_tick_counter = tick_counter;
+            manager.manage_designations(out, false);
+        }
+    }
+    namespace EM = EventManager;
+    if(!has_when<EM::EventHandler>::value){
+        EM::EventHandler tickHandler(onNewHour, hourTicks);
+        EM::registerTick(tickHandler, hourTicks, plugin_self);
+    }
+}
+
+void onNewDay(color_ostream &out, void* tick_ptr){
```

Comment:
In general, formatting is rather inconsistent:
1. `){` versus `) {`
2. `if(` versus `if (`

Reply:
huh?

Reply:
Line 164: `void onNewDay(color_ostream &out, void* tick_ptr){`, no space between the close-paren and the open-brace
Line 196: `void onComplete(color_ostream &out, void* job_ptr) {`, *has* space between close-paren and open-brace

Line 165: `if(enabled && World::isFortressMode()) {`, no space between "if" and open-paren
Line 168: `if ((tick_counter - last_tick_counter) >= dayTicks) {`, *has* space between "if" and open-paren

Both of those variances appear all over the place - it's not necessarily a problem, but it stood out to me so I decided to point it out.

Reply:
Oh, yea. I haven't configured the auto-style rules.